Servers/UI/OJS.Servers.Ui/ClientApp/src/utils/breadcrumb-utils.ts

```diff
@@ -0,0 +1,55 @@
+import
+{
+    BreadcrumbItemNameMaxLength,
+    BreadcrumbItemNameMaxShorteningPercentage,
+    BreadcrumbItemNameMinLength,
+    BreadcrumbItemNameMinShorteningPercentage,
+    Elipsis,
+    MaxBreadcrumbsStringLength,
+} from '../common/constants';
+import { ContestBreadcrumb } from '../common/contest-types';
+
+const calculateBreadcrumbNameSymbolCountToBeRemoved = (nameLength: number) => {
+    /*
+        The calculation is based on a linear interpolation from
+        20% (BreadcrumbItemNameMinShorteningPercentage) to
+        60% (BreadcrumbItemNameMaxShorteningPercentage).
+    */
+    const scale = (nameLength - BreadcrumbItemNameMinLength) / (BreadcrumbItemNameMaxLength - BreadcrumbItemNameMinLength);
+    // eslint-disable-next-line max-len
+    const adjustedPercentage = BreadcrumbItemNameMinShorteningPercentage + scale * (BreadcrumbItemNameMaxShorteningPercentage - BreadcrumbItemNameMinShorteningPercentage);
+
+    return Math.round(nameLength * adjustedPercentage);
+};
+
+const trimBreadcrumbItems = (breadcrumbItems: Array<ContestBreadcrumb>): Array<ContestBreadcrumb> => {
+    // eslint-disable-next-line linebreak-style
```

Comment:
Why did you disable this rule?

Reply:
I forgot to remove it after changing the algorithm's logic.